working/macros/feature-specification.md

```diff
@@ -319,6 +319,63 @@ be constructed, but where you need to build up something in smaller pieces you
 can use the [Fragment][] subtype. Any arbitrary String can be passed to this
 class, allowing you to build up your code fragments however you like.
 
+### Adding New Macro Applications
+
+Macros are allowed to add new macro applications in two ways:
+
+#### Adding Macro Applications to New Declarations
+
+When creating [Code][] instances, a macro may generate code which includes
+macro applications. These macro applications must be from either the current
+phase or a later phase, but cannot be from previous phases.
+
+If a macro application is added which implements an earlier phase, that phase
+is not ran. This should result in a warning if the macro does not also
+implement some phase that will be ran.
```

Comment:
I think it's probably safest to make this an explicit error. I think a macro that defines multiple phases will probably want to be able to reliably assume that its earlier phases have run and that it won't spontaneously pick up in the middle of its phased execution.

Reply:
It am mostly thinking of a macro applying _itself_ to other declarations - granted it could split itself into multiple macros just to avoid the error, but that seems less than ideal.

Consider a macro that runs on a state class, generates a stateful widget class, and also applies itself to some lifecycle methods in the current class. It would likely implement _all three_ phases of macros, but would apply itself to some declarations in the declaration phase. That wouldn't be allowed now and it would have to split itself into multiple macros. Granted macros are easily composable and so it wouldn't be that arduous to do, but it also feels unnecessary.

Reply:
Ah, that's an interesting example, thanks.